tests/unittests/test_fdg_stat.py

```diff
@@ -0,0 +1,413 @@
+                 'allow_more_strict' : True
+                }
+    __salt__ = {}
+
+    def file_stats(name):
+        return {'size': 26, 'group': 'root', 'uid': 0, 'type': 'file', 'mode': '0644', 'gid': 0, 'target': '/etc/passwd',
+                'user': 'root', 'mtime': 1486511757.0, 'atime': 1507221810.408013, 'inode': 1322,
+                'ctime': 1491870657.914388}
+
+    __salt__['file.stats'] = file_stats
+    hubblestack.extmods.fdg.stat.__salt__ = __salt__
+    val = hubblestack.extmods.fdg.stat._check_corner_cases('/etc/file_does_not_exists', expected)
+    assert val[0]
+    assert isinstance(val[1], dict)
+    assert 'Success' in val[1].keys()
+    assert 'expected' in val[1].keys()
+
+
+def test_check_corner_cases_negative_match_on_file_missing():
+    log.info('\n \n Executing test_check_stats_positive_match_on_file_missing')
+    expected = {'mode': '644',
+                          'uid' : 0,
+                          'gid' : 0,
+                          'user' : 'root',
+                          'group' : 'root',
+                          'match_on_file_missing' : False,
+                          'allow_more_strict' : True
+               }
+    __salt__ = {}
+
+    def file_stats(name):
+        return {'size': 26, 'group': 'root', 'uid': 0, 'type': 'file', 'mode': '0644', 'gid': 0, 'target': '/etc/passwd',
+                'user': 'root', 'mtime': 1486511757.0, 'atime': 1507221810.408013, 'inode': 1322,
+                'ctime': 1491870657.914388}
+
+    __salt__['file.stats'] = file_stats
+    hubblestack.extmods.fdg.stat.__salt__ = __salt__
+    val = hubblestack.extmods.fdg.stat._check_corner_cases('/etc/file_does_not_exists', expected)
+    assert not val[0]
+    assert isinstance(val[1], dict)
+    assert 'Failure' in val[1].keys()
+    assert 'expected' in val[1].keys()
+
+
+def test_is_permission_in_limit_positive():
+    log.info('\n \n Executing test_is_permission_in_limit_positive')
+    val = hubblestack.extmods.fdg.stat._is_permission_in_limit(6, 4)
+    assert val
+
+
+def test_is_permission_in_limit_negative():
+    log.info('\n \n Executing test_is_permission_in_limit_negative')
+    val = hubblestack.extmods.fdg.stat._is_permission_in_limit(4, 3)
+    assert not val
+
+
+def test_check_mode_positive():
+    log.info('\n \n Executing test_check_mode_positive')
+    val = hubblestack.extmods.fdg.stat._check_mode("644", "644", False)
+    assert val
+
+
+def test_check_mode_negative():
+    log.info('\n \n Executing test_check_mode_negative')
+    val = hubblestack.extmods.fdg.stat._check_mode("644", "600", False)
+    assert not val
+
+
+def test_check_mode_positive_allow_more_strict():
+    log.info('\n \n Executing test_check_mode_positive_allow_more_strict')
+    val = hubblestack.extmods.fdg.stat._check_mode("644", "600", True)
+    assert val
+
+
+def test_check_mode_negative_allow_more_strict():
+    log.info('\n \n Executing test_check_mode_negative_allow_more_strict')
```

Comment:
please don't put newlines in log lines. I know they're just test logs, but sometimes I view test logs with `--log-cli-level whatever` and it'd just be that much more extra noise.

Reply:
Removed them.